CMakeLists.txt

```diff
@@ -1,87 +1,46 @@
-  elseif(GNUComp)
-    set(c_flags "-ggdb" "-Wall")
-    set(shared_fortran_flags "-ffree-form" "-fbacktrace" "-funroll-loops" "-ggdb" "-Wall") 
-    set(fortran_8_flags "-fdefault-integer-8" "-fdefault-real-8")
-    set(fortran_4_flags)
-  elseif(PGIComp)
-    set(c_flags "-g" "-fast")
-    set(shared_fortran_flags "-g" "-fast")
-    set(fortran_8_flags "-i8" "-r8")
-    set(fortran_4_flags)
-  else()
-    message("unknown compiler!")
-    exit()
-  endif()
+# Ensure at least one of BUILD_SHARED_LIBS and BUILD_STATIC_LIBS is set
+if(NOT (BUILD_STATIC_LIBS OR BUILD_SHARED_LIBS))
+  message(WARNING "Niether BUILD_STATIC_LIBS nor BUILD_SHARED_LIBS is set.  Defaulting to BUILD_STATIC_LIBS=ON")
+  set(BUILD_STATIC_LIBS ON CACHE BOOL "[FORCE ENABLED] Build static libraries" FORCE)
+  set(BUILD_STATIC_LIBS ON)
 endif()
 
-file(GLOB fortran_src ${CMAKE_CURRENT_SOURCE_DIR}/src/*.f)
-file(GLOB c_src ${CMAKE_CURRENT_SOURCE_DIR}/src/*.c)
-set_source_files_properties(${c_src} PROPERTIES COMPILE_OPTIONS "${c_flags}")
-set_source_files_properties(${c_src} PROPERTIES COMPILE_DEFINITIONS "${c_definitions}")
+message(STATUS "Option: BUILD_STATIC_LIBS: ${BUILD_STATIC_LIBS}")
+message(STATUS "Option: BUILD_SHARED_LIBS: ${BUILD_SHARED_LIBS}")
 
-# One target for real32 and one target for real64
-set(kinds "4" "8")
-foreach(kind ${kinds})
-  set(lib_name ${PROJECT_NAME}_${kind})
-  set(versioned_lib_name ${PROJECT_NAME}_v${PROJECT_VERSION}_${kind})
-  add_library(${lib_name} STATIC ${fortran_src} ${c_src})
 
-  set_target_properties(${lib_name} PROPERTIES OUTPUT_NAME "${versioned_lib_name}")
-  
-  set_source_files_properties(${fortran_src} PROPERTIES
-    COMPILE_OPTIONS "${shared_fortran_flags};${fortran_${kind}_flags}")
-  
-  set(module_dir "${CMAKE_CURRENT_BINARY_DIR}/include_${kind}")
-  set_target_properties(${lib_name} PROPERTIES Fortran_MODULE_DIRECTORY "${module_dir}")
+if(NOT CMAKE_BUILD_TYPE MATCHES "^(Debug|Release|RelWithDebInfo|MinSizeRel)$")
+  message(STATUS "Setting build type to 'Release' as none was specified.")
+  set(CMAKE_BUILD_TYPE
+      "Release"
+      CACHE STRING "Choose the type of build." FORCE)
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release"
+                                               "MinSizeRel" "RelWithDebInfo")
+endif()
 
-  target_include_directories(${lib_name} PUBLIC
-    $<BUILD_INTERFACE:${module_dir}>
-    $<INSTALL_INTERFACE:${CMAKE_INSTALL_PREFIX}/include_${kind}>)
+if(NOT CMAKE_C_COMPILER_ID MATCHES "^(Intel|GNU|Clang|AppleClang)$")
+  message(WARNING "Compiler not officially supported: ${CMAKE_C_COMPILER_ID}")
+endif()
 
-  install(TARGETS ${lib_name}
-    EXPORT ${PROJECT_NAME}-config
-    RUNTIME DESTINATION bin
-    LIBRARY DESTINATION lib
-    ARCHIVE DESTINATION lib) 
-  install(DIRECTORY ${module_dir} DESTINATION ${CMAKE_INSTALL_PREFIX})
-endforeach()
+add_subdirectory(src)
 
-install(EXPORT ${PROJECT_NAME}-config DESTINATION ${CMAKE_INSTALL_PREFIX})
+if (${ENABLE_TESTS})
```

Comment:
Should this really be an option? 
IMO, tests should always be enabled.  A user may or may not run them.

But, if you really want to make them optional, this should be a cmake `OPTION()` supplied with `-DENABLE_TESTS=ON`

Reply:
Also, you should add somewhere at the top
```
option(ENABLE_TESTS "Enable testing with pFUnit" OFF)
```

Reply:
1. I did that because I didn't want to force pFUnit. Maybe a if pfunit isn't found then don't build them? Or just require it?

2. I think you're right

Reply:
Fixed